Simplenote/src/androidTest/java/com/automattic/simplenote/integration/TagDialogFragmentTest.kt

```diff
@@ -0,0 +1,326 @@
+        val cancelText = getResourceString(R.string.cancel)
+        onView(withText(cancelText)).inRoot(isDialog()).perform(click())
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(note.tags, listOf("tag1"))
+    }
+
+    @Test
+    fun editTagWithSpace() {
+        createTag("tag10")
+        // To edit tags, tags should belong a note
+        val note = createNote("Hello World", listOf("tag10"))
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(notesBucket.count(), 1)
+
+        launchFragment("tag10")
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText("tag 3"))
+
+        val tagWithSpaceMessage = getResourceString(R.string.tag_error_spaces)
+        onView(withId(R.id.input_tag_name)).check(matches(hasTextInputLayoutErrorText(tagWithSpaceMessage)))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(not(isEnabled())))
+        val cancelText = getResourceString(R.string.cancel)
+        onView(withText(cancelText)).inRoot(isDialog()).perform(click())
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(note.tags, listOf("tag10"))
+    }
+
+    @Test
+    fun editTagWithEmpty() {
+        createTag("tag10")
+        // To edit tags, tags should belong a note
+        val note = createNote("Hello World", listOf("tag10"))
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(notesBucket.count(), 1)
+
+        launchFragment("tag10")
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText(""))
+
+        val tagEmpty = getResourceString(R.string.tag_error_empty)
+        onView(withId(R.id.input_tag_name)).check(matches(hasTextInputLayoutErrorText(tagEmpty)))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(not(isEnabled())))
+        val cancelText = getResourceString(R.string.cancel)
+        onView(withText(cancelText)).inRoot(isDialog()).perform(click())
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(note.tags, listOf("tag10"))
+    }
+
+    @Test
+    fun editTagInMultipleNotes() {
+        createTag("tag1")
+        createTag("tag2")
+        createTag("tag3")
+        // To edit tags, tags should belong a note
+        val note1 = createNote("Hello1", listOf("tag1", "tag2"))
+        val note2 = createNote("Hello1", listOf("tag2", "tag3"))
+        val note3 = createNote("Hello1", listOf("tag1", "tag3"))
```

Comment:
Minor (🔍): Reader the test cases within [1364](https://github.com/Automattic/simplenote-android/issues/1364) issue I am seeing that the note names are `Hello1`, `Hello2` and `Hello3`. I know it doesn't matter much, but maybe it is good to have different notes and align those with the test cases issue. Wdyt?

PS: Same applies everywhere else.

Reply:
Fixed in 3bfd5bda11dc003ae4f9588e849af888c801f979.